pkg/cmd/resource/namespace_test.go

```diff
@@ -0,0 +1,29 @@
+package resource
+
+import (
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewNamespaceCmd_NonEmptyName(t *testing.T) {
+	rootCmd := &cobra.Command{Annotations: make(map[string]string)}
+
+	nsc := NewNamespaceCmd(rootCmd, "foo")
+
+	assert.Equal(t, "foo", nsc.Name)
+	assert.True(t, rootCmd.HasSubCommands())
+	assert.Equal(t, "namespace", rootCmd.Annotations["foo"])
+}
+
+func TestNewNamespaceCmd_EmptyName(t *testing.T) {
+	rootCmd := &cobra.Command{Annotations: make(map[string]string)}
+
+	nsc := NewNamespaceCmd(rootCmd, "")
+
+	assert.Equal(t, "", nsc.Name)
+	assert.False(t, rootCmd.HasSubCommands())
+	_, ok := rootCmd.Annotations[""]
+	assert.False(t, ok)
+}
```

Comment:
All the template stuff is evaluated at runtime only, so you may want to add a test here that exercises that in at least some basic way to make sure that it's right (and same goes for resource and operation commands).

Reply:
👍